ia32/cmd.c

```diff
@@ -155,6 +157,39 @@ char *cmd_getnext(char *line, unsigned int *pos, char *blanks, char *cites, char
 }
 
 
+/* Function parses device name and returns its number */
+static unsigned int cmd_getdevice(char *line, unsigned int *pos, char *blanks, char *cites, char *word, unsigned int len)
+{
+	unsigned int dn;
+
+	if (cmd_getnext(line, pos, blanks, cites, word, len) == NULL)
+		return ERR_ARG;
+
+	for (dn = 0; devices[dn].name; dn++) {
+		if (!plostd_strcmp(word, devices[dn].name))
+			break;
+	}
+
+	if (!devices[dn].name)
+		return ERR_ARG;
+
+	return dn;
+}
+
+
+/* Function parses file name / LBA and returns its handle */
+static s32 cmd_gethandle(unsigned int dn, char *line, unsigned int *pos, char *blanks, char *cites, char *word, unsigned int len)
+{
+	if (cmd_getnext(line, pos, blanks, cites, word, len) == NULL)
+		return ERR_ARG;
+
+	if (*word == '\0')
+		return ERR_ARG;
+
+	return (dn >= PDN_COM1 && dn <= PDN_COM2) ? phfs_open(devices[dn].pdn, word, 0) : (s32)plostd_atol(word);
```

Comment:
This line could be replaced by just `return phfs_open(devices[dn].pdn, word, 0);` if disk_open() was implemented like so:
```
s32 disk_open(u16 dn, char *name, u32 flags)
{
    return plostd_atol(word);
}
```